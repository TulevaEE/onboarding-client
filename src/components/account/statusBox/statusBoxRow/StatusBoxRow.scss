@import '../../../variables';

.status-box-row li {
  list-style: none;
}

.status-box-row ul {
  padding: 0;
  margin-bottom: 0;
}

.status-box-row .btn {
  min-width: 161px;
}

.fa.fa-check {
  color: $green;
}

.fa.fa-times {
  color: $red;
}

.fa.fa-exclamation-triangle {
  color: $yellow-dark;
}
