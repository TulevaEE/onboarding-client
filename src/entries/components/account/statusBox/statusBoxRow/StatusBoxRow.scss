@import "../../../variables";

li.status-box-row {
  list-style: none;
}

ul.status-box-row {
  padding: 0;
  margin-bottom: 0;
}

.fa.fa-check {
  color: $green;
}

.fa.fa-times {
  color: $red;
}
