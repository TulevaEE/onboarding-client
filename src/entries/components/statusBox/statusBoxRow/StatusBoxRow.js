import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { Message } from 'retranslate';

const StatusBoxRow = ({ name, showAction, ok }) => {
  const displayName = <Message>{name}</Message>;
  const displayAction = 'XXXXXXX tbd siia nupp';
  return (
    <div className="row tv-table__row py-2">
      <div className="col-12 col-sm">
        {ok ? 'âœ”' : 'ðŸ—™'} <b>{displayName}</b>
      </div>
      <div className="col-12 col-sm text-sm-right">{showAction ? <b>{displayAction}</b> : ''}</div>
    </div>
  );
};

StatusBoxRow.defaultProps = {
  name: '',
  showAction: false,
  ok: false,
};

StatusBoxRow.propTypes = {
  name: Types.oneOfType([Types.node, Types.string]),
  showAction: Types.bool,
  ok: Types.bool,
};

export default StatusBoxRow;
