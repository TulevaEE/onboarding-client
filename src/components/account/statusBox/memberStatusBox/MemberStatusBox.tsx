import React from 'react';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import StatusBoxRow from '../statusBoxRow';

interface Props {
  loading: boolean;
  memberNumber: number | null;
}

export const MemberStatusBox: React.FunctionComponent<Props> = ({
  loading = false,
  memberNumber,
}) => {
  const isTulevaMember = memberNumber != null;
  const tulevaData = isTulevaMember
    ? [<FormattedMessage id="account.member.statement" values={{ memberNumber }} />]
    : [<FormattedMessage id="account.non.member.statement" />];

  return (
    <StatusBoxRow
      last
      ok={isTulevaMember}
      warning={!isTulevaMember}
      showAction={!loading}
      name={<FormattedMessage id="account.status.choice.tuleva" />}
      lines={tulevaData}
    >
      {!isTulevaMember && (
        <Link to="/join" className="btn btn-light">
          <FormattedMessage id="account.status.choice.join.tuleva" />
        </Link>
      )}
    </StatusBoxRow>
  );
};

type State = {
  login: {
    loadingUserConversion: boolean;
    user: { memberNumber: number };
  };
};

const mapStateToProps = (state: State) => ({
  memberNumber: (state.login.user || {}).memberNumber,
  loading: state.login.loadingUserConversion,
});

export default connect(mapStateToProps)(MemberStatusBox);
