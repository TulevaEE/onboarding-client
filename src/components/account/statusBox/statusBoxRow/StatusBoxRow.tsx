import React from 'react';
import './StatusBoxRow.scss';

const CheckMark: React.FunctionComponent<{
  checked?: boolean;
  warning?: boolean;
  error?: boolean;
}> = ({ checked = false, warning = false, error = false }) => (
  <div
    className={`ml-3 mr-2 fa${checked ? ' fa-check' : ''}${error ? ' fa-times' : ''}${
      warning ? ' fa-exclamation-triangle' : ''
    }`}
  />
);

export const StatusBoxRow: React.FunctionComponent<{
  name?: React.ReactNode;
  lines?: React.ReactNode[];
  showAction?: boolean;
  ok?: boolean;
  warning?: boolean;
  error?: boolean;
  last?: boolean;
  children?: React.ReactNode;
}> = ({
  name = '',
  lines = [],
  showAction = false,
  ok = false,
  warning = false,
  error = false,
  children = '',
  last = false,
}) => {
  const formattedLines = (
    <ul>
      {lines &&
        lines.length > 0 &&
        lines.map((line, i) => {
          return (
            <li className="pl-2" key={i}>
              {line}
            </li>
          );
        })}
    </ul>
  );
  return (
    <div
      className={`d-flex flex-sm-row flex-column justify-content-between py-2 status-box-row ${
        !last ? 'tv-table__row' : ''
      }`}
    >
      <div className="d-flex">
        <div className="d-flex flex-column justify-content-center">
          <CheckMark checked={ok} warning={warning} error={error} />
        </div>
        <div className="d-flex flex-column justify-content-center">
          <div className="mt-0 pt-1 pl-2">
            <b>{name}</b>
          </div>
          {formattedLines}
        </div>
      </div>
      <div className="d-flex flex-column justify-content-center my-2 mx-3 text-nowrap">
        {showAction && children ? children : ''}
      </div>
    </div>
  );
};

export default StatusBoxRow;
