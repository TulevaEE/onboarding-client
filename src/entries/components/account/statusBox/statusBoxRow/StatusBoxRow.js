import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { Message } from 'retranslate';
import './StatusBoxRow.scss';

const CheckMark = ({ checked }) => (
  <div className={`ml-3 mr-2 fa${checked ? ' fa-check' : ' fa-times'}`} />
);
CheckMark.defaultProps = {
  checked: false,
};

CheckMark.propTypes = {
  checked: Types.bool,
};

const StatusBoxRow = ({ name, lines, showAction, ok, children, last }) => {
  const displayName = <Message>{name}</Message>;
  const formattedLines = (
    <ul className="status-box-row">
      {lines &&
        lines.length > 0 &&
        lines.map(line => {
          return (
            <li className="status-box-row pl-2" key={line}>
              {line}
            </li>
          );
        })}
    </ul>
  );
  return (
    <div className={`d-flex justify-content-between py-2 ${!last ? 'tv-table__row' : ''}`}>
      <div className="d-flex">
        <div className="d-flex flex-column justify-content-center">
          <CheckMark checked={ok} />
        </div>
        <div className="d-flex flex-column justify-content-center">
          <div className="mt-0 pt-1 pl-2">
            <b>{displayName}</b>
          </div>
          {formattedLines}
        </div>
      </div>
      <div className="d-flex flex-column justify-content-center mx-3 text-nowrap">
        {showAction && children ? children : ''}
      </div>
    </div>
  );
};

StatusBoxRow.defaultProps = {
  name: '',
  lines: [],
  showAction: false,
  ok: false,
  children: null,
  last: false,
};

StatusBoxRow.propTypes = {
  name: Types.oneOfType([Types.node, Types.string]),
  lines: Types.arrayOf(Types.oneOfType([Types.node, Types.string, Types.string])),
  showAction: Types.bool,
  ok: Types.bool,
  children: Types.node,
  last: Types.bool,
};

export default StatusBoxRow;
