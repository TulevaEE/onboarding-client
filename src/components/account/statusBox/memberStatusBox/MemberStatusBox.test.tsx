import React from 'react';
import { shallow } from 'enzyme';
import { MemberStatusBox } from './MemberStatusBox';
import StatusBoxRow from '../statusBoxRow';

describe('MemberStatusBox', () => {
  const loading = false;
  const memberNumber = 123;
  const memberJoinDate = '2024-02-16T12:15:10Z';
  const component = shallow(
    <MemberStatusBox
      loading={loading}
      memberNumber={memberNumber}
      memberJoinDate={memberJoinDate}
    />,
  );

  it('renders the full component', () => {
    expect(component).toMatchSnapshot();
  });

  it('does not show action when still loading', () => {
    component.setProps({ loading: true });
    expect(component.find(StatusBoxRow).prop('showAction')).toBeFalsy();
  });

  it('offers to join when not member', () => {
    component.setProps({ memberNumber: null });
    expect(component).toMatchSnapshot();
  });
});
