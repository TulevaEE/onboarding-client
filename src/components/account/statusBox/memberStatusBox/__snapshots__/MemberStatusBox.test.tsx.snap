// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemberStatusBox offers to join when not member 1`] = `
<StatusBoxRow
  last={true}
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.non.member.statement"
      />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.tuleva"
    />
  }
  ok={false}
  showAction={false}
  warning={true}
>
  <Link
    className="btn btn-light"
    to="/join"
  >
    <MemoizedFormattedMessage
      id="account.status.choice.join.tuleva"
    />
  </Link>
</StatusBoxRow>
`;

exports[`MemberStatusBox renders the full component 1`] = `
<StatusBoxRow
  last={true}
  lines={
    Array [
      <Memo(MemoizedFormattedMessage)
        id="account.member.statement"
        values={
          Object {
            "memberNumber": 123,
          }
        }
      />,
    ]
  }
  name={
    <Memo(MemoizedFormattedMessage)
      id="account.status.choice.tuleva"
    />
  }
  ok={true}
  showAction={true}
  warning={false}
/>
`;
